extension/src/background/messageListener/lyraApiMessageListener.ts

```diff
@@ -65,18 +67,30 @@ export const lyraApiMessageListener = (
   };
 
   const submitTransaction = () => {
-    const { transactionXdr } = request;
+    let isDomainListedAllowed = false;
 
+    const { transactionXdr } = request;
     const transaction = StellarSdk.TransactionBuilder.fromXDR(
       transactionXdr,
       StellarSdk.Networks[NETWORK],
     );
 
     const { tab } = sender;
+    const tabUrl = tab?.url ? tab.url : "";
+    const domain = getUrlHostname(tabUrl);
+    const punycodedDomain = getPunycodedDomain(domain);
+
+    const allowListStr = localStorage.getItem(ALLOWLIST_ID) || "";
+    const allowList = allowListStr.split(",");
+
+    if (allowList.includes(punycodedDomain)) {
+      isDomainListedAllowed = true;
+    }
```

Comment:
super small, but it's a little less error prone to do:
```
isDomainListedAllowed = allowList.includes(punycodedDomain);
```

Reply:
Updated, @piyalbasu